File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.java
class Solution {
    public int longestBeautifulSubstring(String word) {
        HashSet<Character> set=new HashSet<>();
        int left=0, right=0, max=0;
        while(right<word.length()){
            if(right==0 || right>0 && word.charAt(right)>=word.charAt(right-1)){
                set.add(word.charAt(right));
            }else{
                if(set.size()==5){
                    max=Math.max(max, right-left);
                }
                set.clear();
                left=right;
                set.add(word.charAt(right));
            }
            right++;
        }
        if(set.size()==5) max=Math.max(max, word.length() - left);
        return max;
    }
}

File: 1839-longest-substring-of-all-vowels-in-order/1839-longest-substring-of-all-vowels-in-order.py
class Solution:
    def longestBeautifulSubstring(self, word: str) -> int:
        res = 0
        seen = set()
        left = -1
        right = 0
        while right<len(word):
            if right>0 and word[right]<word[right-1]:
                seen.clear()
                left = right-1
            seen.add(word[right])
            if len(seen) == 5:
                res = max(res, right-left)
            right += 1
        return res